### PR TITLE
Adds a new weapon choice for the crew

### PR DESCRIPTION
**Pull request by KaiserBlackwood**
## About The Pull Request

Adds a new weapon to the game's already expansive roster, but it's available to security via cargo's Armory tab. This nifty little reuse of assets (sprite change pending, as I still haven't found a good spriter to help me with that) allows players to have an energy alternative to CMG without sacrificing rate of fire.

## How This Contributes To The Skyrat Roleplay Experience

This PR offers crew an additional choice of weapon to broaden their possibilities. While yes, being an addition mostly oriented towards security, this gun can be a perfect conversation-starter for gun-nuts. It does come with its own manufacturer as well, lore of which I intend on expanding in case this PR does get accepted. For the most part, it's just a thing to add variety for players, because being pinned between CMG, MCR and NRI guns as your choice for a rifle-sized firearm isn't very inspiring, if you ask me. Also, it's a pretty futuristic weapon of choice.

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
  
Here are some videos and images that prove that this gun _indeed_ works as intended.

image
https://github.com/Skyrat-SS13/Skyrat-tg/assets/69368878/10708fc6-c401-4875-8b88-bfb870d0624d
https://github.com/Skyrat-SS13/Skyrat-tg/assets/69368878/b95f30dd-3660-486a-a37b-48bc0db3497c
image

</details>

## Changelog

:cl:
add: Added new weapon crate to Cargo's Armory tab: **NTX-867 Starfall**
balance: Added an "automatic" energy weapon for security, which can be a game-changer in some cases.
image: Added multiple sprites for SMG's proprietary batteries: Red stripe for Lethal, Yellow stripe for Non-Lethal.
/:cl:

## Notes

SMG deals **20 damage per projectile**, meaning that with one burst you can put around 40 stam/energy damage in the target if you land both shots. The fire delay should be long enough to allow you to dodge 2nd shot of the burst.

Each magazine holds **20 rounds**, but during reloads, if your magazine wasn't empty, you may lose 1 shot due to how chambering works. I am working on a way to solve this. Each mag can be recharged in the gun-charger, so it shouldn't be a big problem.

### CODE REVIEW

**Comment by nikothedude** [comment]
Aren't we operating on the idea of energy-based guns not being mag-fed? Or at least the guns we expect security to be holding? MCRs got that removed a while ago.

**Comment by KaiserBlackwood** [comment]
I can adjust that, tbh. Just will take a while longer. Mainly I did that because it was the easiest way to make an automatic energy gun.

File: modular_skyrat/modules/modular_weapons/code/energy.dm [hunk 1]
```diff
@@ -72,6 +72,83 @@
...
+	desc = "A reliable and rechargable battery that serves as an energy source for NTX-867 Starfall."
+	icon_state = "laser_smg_L-20"
+	base_icon_state = "laser_smg_L"
+	ammo_type = /obj/item/ammo_casing/caseless/laser_smg/lethal
+	caliber = CALIBER_LASER
+	max_ammo = 20
+
+/obj/item/ammo_box/magazine/recharge/laser_smg/lethal/update_desc()
+	. = ..()
+	desc = "[initial(desc)] It has [stored_ammo.len] shot\s left."
```
Comment (softcerv) [inline modular_skyrat/modules/modular_weapons/code/energy.dm:119]: Wouldn't this overwrite the current description for the item?

File: modular_skyrat/modules/modular_weapons/code/energy.dm [hunk 2]
```diff
@@ -72,6 +72,83 @@
...
+	desc = "A reliable and rechargable battery that serves as an energy source for NTX-867 Starfall."
+	icon_state = "laser_smg_NL-20"
+	base_icon_state = "laser_smg_NL"
+	ammo_type = /obj/item/ammo_casing/caseless/laser_smg/nonlethal
+	caliber = CALIBER_LASER
+	max_ammo = 20
+
+/obj/item/ammo_box/magazine/recharge/laser_smg/nonlethal/update_desc()
+	. = ..()
+	desc = "[initial(desc)] It has [stored_ammo.len] shot\s left."
```
Comment (softcerv) [inline modular_skyrat/modules/modular_weapons/code/energy.dm:141]: same with this.

File: modular_skyrat/modules/modular_weapons/code/energy.dm [hunk 3]
```diff
@@ -72,6 +72,83 @@
...
+
+/obj/item/ammo_box/magazine/recharge/laser_smg/nonlethal/update_icon_state()
+	. = ..()
+	icon_state = "[base_icon_state]-[round(ammo_count(), 4)]"
+
+/obj/item/ammo_box/magazine/recharge/laser_smg/nonlethal/attack_self() //No popping out the "bullets"
+	return
```
Comment (softcerv) [inline modular_skyrat/modules/modular_weapons/code/energy.dm:148]: Same here

File: modular_skyrat/modules/modular_weapons/code/energy.dm [hunk 4]
```diff
@@ -160,6 +237,23 @@
 	stamina = 30
 	range = 6
 
+// NTX-867 Non-lethal Projectile //
+
+/obj/item/ammo_casing/energy/electrode/laser_smg
+	projectile_type = /obj/projectile/energy/electrode/knockdown
```
Comment (softcerv) [inline modular_skyrat/modules/modular_weapons/code/energy.dm:243]: wait, are these taser projectiles?

File: modular_skyrat/modules/modular_weapons/code/energy.dm [hunk 5]
```diff
@@ -248,3 +362,22 @@
...
+
+/obj/item/ammo_casing/caseless/laser_smg/nonlethal
+	name = "type V2 energy projectile"
+	desc = "An energy cell that once triggered, creates a non-lethal projectile, melting it's own casing in the process."
+	icon = 'modular_skyrat/modules/modular_weapons/icons/obj/ammo.dmi'
+	icon_state = "disabler_shell"
+	worn_icon_state = "shell"
+	caliber = "electrode"
+	custom_materials = list(/datum/material/iron=4000,/datum/material/plasma=250)
+	projectile_type = /obj/projectile/energy/electrode/laser_smg
```
Comment (softcerv) [inline modular_skyrat/modules/modular_weapons/code/energy.dm:383]: You could use inheritance to reduce a lot of the code needed here. Please send me a message on discord if you need help with this.

File: modular_skyrat/modules/sec_haul/code/guns/gunsets.dm [hunk 6]
```diff
@@ -439,6 +439,25 @@
 	. = ..()
 	new /obj/item/gun/energy/e_gun(src)
 
+// NX-867 Starfall //
+
+/obj/item/storage/box/gunset/laser_smg
+	name = "NX-867 gun case"
+	desc = "A briefcase that contains NX-867 Starfall SMG. Comes with batteries."
+	w_class = WEIGHT_CLASS_BULKY
```
Comment (softcerv) [inline modular_skyrat/modules/sec_haul/code/guns/gunsets.dm:447]:
```suggestion
	w_class = WEIGHT_CLASS_BULKY
	
```

File: modular_skyrat/modules/modular_weapons/code/energy.dm [hunk 7]
```diff
@@ -72,6 +72,83 @@
...
+	bolt_type = BOLT_TYPE_OPEN
+	show_bolt_icon = FALSE
+
+/obj/item/gun/ballistic/automatic/laser_smg/Initialize(mapload)
+	. = ..()
+	bolt_locked = FALSE
+
+/obj/item/gun/ballistic/automatic/laser_smg/insert_magazine(mob/user, obj/item/ammo_box/magazine/AM, display_message)
+	. = ..()
+
```
Comment (softcerv) [inline modular_skyrat/modules/modular_weapons/code/energy.dm:102]:
```suggestion
```

File: modular_skyrat/modules/modular_weapons/code/energy.dm [hunk 8]
```diff
@@ -72,6 +72,83 @@
...
+/obj/item/ammo_box/magazine/recharge/laser_smg/lethal/update_desc()
+	. = ..()
+	desc = "[initial(desc)] It has [stored_ammo.len] shot\s left."
+
+/obj/item/ammo_box/magazine/recharge/laser_smg/lethal/update_icon_state()
+	. = ..()
+	icon_state = "[base_icon_state]-[round(ammo_count(), 4)]"
+
+/obj/item/ammo_box/magazine/recharge/laser_smg/lethal/attack_self() //No popping out the "bullets"
+	return
```
Comment (softcerv) [inline modular_skyrat/modules/modular_weapons/code/energy.dm:126]: When doing this does the code runtime at all? I know for some of the attack procs, you need to return with a certain define, but I am unsure if this is one of them.

**Review by softcerv — COMMENTED** [review]
some things I've noticed

File: modular_skyrat/modules/modular_weapons/code/energy.dm (hunk 1 above)
Comment (KaiserBlackwood) [reply modular_skyrat/modules/modular_weapons/code/energy.dm:119]: Nope, it doesn't. It works perfectly fine.

File: modular_skyrat/modules/modular_weapons/code/energy.dm (hunk 4 above)
Comment (KaiserBlackwood) [reply modular_skyrat/modules/modular_weapons/code/energy.dm:243]: "taser" projectiles. They cause sprite jitters and stuttering, but beyond that no impact.

File: modular_skyrat/modules/modular_weapons/code/energy.dm (hunk 8 above)
Comment (KaiserBlackwood) [reply modular_skyrat/modules/modular_weapons/code/energy.dm:126]: nnno, I don't think so. I've taken that from the rechargeable magazine from original TG code.

File: modular_skyrat/modules/modular_weapons/code/energy.dm (hunk 5 above)
Comment (KaiserBlackwood) [reply modular_skyrat/modules/modular_weapons/code/energy.dm:383]: I don't have your discord, so if you'll send it to me, then sure.

File: modular_skyrat/modules/modular_weapons/code/energy.dm (hunk 1 above)
Comment (softcerv) [reply modular_skyrat/modules/modular_weapons/code/energy.dm:119]: ah, I think I missed something when reading this then. My bad.

File: modular_skyrat/modules/modular_weapons/code/energy.dm (hunk 4 above)
Comment (softcerv) [reply modular_skyrat/modules/modular_weapons/code/energy.dm:243]:
```
/obj/projectile/energy/electrode/knockdown
	name = "electrobolt"
	icon = 'modular_skyrat/modules/modular_weapons/icons/obj/projectiles.dmi'
	icon_state = "electro_bolt"
	knockdown = 50
	stamina = 30
```
are you positive about that? It says in the code for the projectile that it has knockdown.

**Comment by HoodMilk** [comment]
These are cool.

File: modular_skyrat/modules/modular_weapons/code/energy.dm (hunk 4 above)
Comment (chemistrymain2) [reply modular_skyrat/modules/modular_weapons/code/energy.dm:243]: You appear to be using projectile/energy/electrode/knockdown and not projectile/energy/electrode/laser_smg.

**Comment by chemistrymain2** [comment]
Neat idea, but the burstfire is still concerning, and the laser bolt has a different travel speed to the electrode bolt.

Another honest question is 'what does this do that the MCR doesn't?' It's a neat sprite, but if it has no other functionality other than being 'this gun but different and we have a burstfire ig' then it really does fall under the same sort of bloat the sabel did.

**Comment by KaiserBlackwood** [comment]
> 

I am working out how to implement my initial ideas into the code. Also what it does that MCR doesn't?
Well, firstly, my personal feeling is that MCR sprite is not doing the gun any favors, and I don't enjoy it. Secondly, MCR isn't the type of gun I want to use when I play security of 26th century. Yes, it's mostly purely aesthetic reasons why I add this. I am still working around the burst-fire/other modes I want this gun to fire in and sprite gonna be changed to something completely different (as mostly going to be the gun). As for "bloat", despite us having like 30-40 primaries, we use like 4 of them on-station. Funnily enough, it's because most of them aren't fitting with server's direction. Therefore, I add a gun that is fitting. At least in my opinion.

**Pull request closed without merging**